examples/shielded_rewards.rs

```diff
@@ -0,0 +1,289 @@
+    let lock_target = lock_target
+        .increase_precision(incent_decimals.into())
+        .map_err(std::io::Error::other)?;
+
+    // Let M be the desired minimum amount of TOK required to get rewards
+    print!("Incentivisation threshold in {}: ", incent_code);
+    std::io::stdout().flush()?;
+    let mut incent_threshold = String::new();
+    stdin.read_line(&mut incent_threshold)?;
+    let incent_threshold = incent_threshold
+        .trim()
+        .parse::<DenominatedAmount>()
+        .map_err(std::io::Error::other)?;
+    let incent_threshold = incent_threshold
+        .increase_precision(incent_decimals.into())
+        .map_err(std::io::Error::other)?;
+
+    // It must be the case that X/I <= M. Or equivalently I >= X/M.
+    let min_inflation = lock_target.amount().raw_amount()
+        / incent_threshold.amount().raw_amount();
+    let min_inflation =
+        DenominatedAmount::new(min_inflation.into(), native_decimals.into());
+    println!(
+        "Inflation must be more than {} {} to realize non-zero rewards for \
+         users holding more than {} {} in a pool holding {} {}.",
+        min_inflation,
+        native_code,
+        incent_threshold,
+        incent_code,
+        lock_target,
+        incent_code
+    );
+    print!("Inflation (>= {}) in {}: ", min_inflation, native_code);
+    std::io::stdout().flush()?;
+
+    // Let I be the computed amount of uNAM inflation in a single round of the
+    // mechanism due to the incentivised token TOK.
+    let mut inflation = String::new();
+    stdin.read_line(&mut inflation)?;
+    let inflation = inflation
+        .trim()
+        .parse::<DenominatedAmount>()
+        .map_err(std::io::Error::other)?;
+    let inflation = inflation
+        .increase_precision(native_decimals.into())
+        .map_err(std::io::Error::other)?;
+
+    // A necessary condition for there to be inflation is that floor(I*P/X)>=1.
+    let min_precision =
+        lock_target.amount().raw_amount() / inflation.amount().raw_amount();
+    // A necessary condition for users holding more than the threshold to get
+    // rewards is that P<=M.
+    let max_precision = incent_threshold.amount().raw_amount();
+    println!(
+        "At an inflation of {} {}, the precision must be between {} and {} in \
+         order to realize non-zero rewards.",
+        inflation, native_code, min_precision, max_precision
+    );
+
+    // Let P be the precision of the token TOK
+    print!("Precision ([{}, {}]): ", min_precision, max_precision);
+    std::io::stdout().flush()?;
+    let mut precision = String::new();
+    stdin.read_line(&mut precision)?;
+    let precision = precision
+        .trim()
+        .parse::<Dec>()
+        .map_err(std::io::Error::other)?;
+    println!(
+        "Users holding at least {} {} in the shielded pool will receive \
+         rewards.",
+        precision, incent_code
+    );
+
+    // It must be that S*C/Y >= I, which implies that the reward rate
```

Comment:
`C` is defined below. Consider moving doc string up. Also, `S` seems to be missing a definition in the doc strings entirely.

Reply:
Thanks. I think `S` is defined at https://github.com/anoma/namada/blob/1cb9fccc857d2c0608f01a01fb72518bda9e5b2b/examples/shielded_rewards.rs#L81 .